src/model/cloud-runner/aws/aws-task-runner.ts

```diff
@@ -0,0 +1,216 @@
+    let readingLogs = true;
+    let timestamp: number = 0;
+    while (readingLogs) {
+      await new Promise((resolve) => setTimeout(resolve, 1500));
+      const taskData = await AWSTaskRunner.describeTasks(ECS, clusterName, taskArn);
+      ({ timestamp, readingLogs } = AWSTaskRunner.checkStreamingShouldContinue(taskData, timestamp, readingLogs));
+      ({ iterator, readingLogs } = await AWSTaskRunner.handleLogStreamIteration(
+        kinesis,
+        iterator,
+        readingLogs,
+        taskDef,
+      ));
+    }
+  }
+
+  private static async handleLogStreamIteration(
+    kinesis: AWS.Kinesis,
+    iterator: string,
+    readingLogs: boolean,
+    taskDef: CloudRunnerTaskDef,
+  ) {
+    const records = await kinesis
+      .getRecords({
+        ShardIterator: iterator,
+      })
+      .promise();
+    iterator = records.NextShardIterator || '';
+    readingLogs = AWSTaskRunner.logRecords(records, iterator, taskDef, readingLogs);
+    return { iterator, readingLogs };
+  }
+
+  private static checkStreamingShouldContinue(taskData: AWS.ECS.Task, timestamp: number, readingLogs: boolean) {
+    if (taskData?.lastStatus !== 'RUNNING') {
+      if (timestamp === 0) {
+        CloudRunnerLogger.log('## Cloud runner job stopped, streaming end of logs');
+        timestamp = Date.now();
+      }
+      if (timestamp !== 0 && Date.now() - timestamp > 30000) {
+        CloudRunnerLogger.log('## Cloud runner status is not RUNNING for 30 seconds, last query for logs');
+        readingLogs = false;
+      }
+      CloudRunnerLogger.log(`## Status of job: ${taskData.lastStatus}`);
+    }
+    return { timestamp, readingLogs };
+  }
+
+  private static logRecords(records, iterator: string, taskDef: CloudRunnerTaskDef, readingLogs: boolean) {
+    if (records.Records.length > 0 && iterator) {
+      for (let index = 0; index < records.Records.length; index++) {
+        const json = JSON.parse(
+          zlib.gunzipSync(Buffer.from(records.Records[index].Data as string, 'base64')).toString('utf8'),
+        );
+        if (json.messageType === 'DATA_MESSAGE') {
+          for (let logEventsIndex = 0; logEventsIndex < json.logEvents.length; logEventsIndex++) {
+            let message = json.logEvents[logEventsIndex].message;
+            if (json.logEvents[logEventsIndex].message.includes(taskDef.logid)) {
+              CloudRunnerLogger.log('End of log transmission received');
+              readingLogs = false;
+            } else if (message.includes('Rebuilding Library because the asset database could not be found!')) {
+              core.warning('LIBRARY NOT FOUND!');
+            }
+            message = `[${CloudRunnerStatics.logPrefix}] ${message}`;
+            if (CloudRunnerState.buildParams.logToFile) {
+              fs.appendFileSync(`${CloudRunnerState.buildGuid}-outputfile.txt`, `${message}\r\n`);
+            }
+            CloudRunnerLogger.log(message);
+          }
+        }
+      }
+    }
+    return readingLogs;
```

Comment:
Not sure about signature of this method. But I guess fine for this iteration.

Reply:
Will review